test/manipulation.test.js

```diff
@@ -768,6 +768,86 @@ describe('manipulation', function() {
     });
   });
 
+  describe('upsertWithWhere', function() {
+    var ds = getSchema();
+    var Post;
+    before('prepare "Post" model', function(done) {
+      Post = ds.define('Post', {
+        empid: { type: Number, id: true },
+        name: { type: String },
+        city: { type: String },
+      });
+      ds.automigrate('Post', done);
+    });
+
+    it('has an alias "patchOrCreateWithWhere"', function() {
+      StubUser.upsertWithWhere.should.equal(StubUser.patchOrCreateWithWhere);
+    });
+
+    it('fails the upsertWithWhere operation when data object is empty', function(done) {
+      options = {};
+      Post.upsertWithWhere({ name: 'John Lennon' }, {}, options,
```

Comment:
A `Post` called `John Lennon`, that looks strange to me. Perhaps the model should be called `Person` instead?